02_knight_and_princess/robert_gaisin/main.cpp

```diff
@@ -0,0 +1,235 @@
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+using VertexId = int;
+using EdgeId = int;
+
+std::ostream& operator<<(std::ostream& out, const vector<int>& int_vector) {
+  for (auto it = int_vector.begin(); it != int_vector.end(); ++it) {
+    out << *it;
+    if (it != int_vector.end() - 1) {
+      out << ", ";
+    }
+  }
+  return out;
+}
+
+struct Vertex {
+  VertexId id = 0;
+  int depth = 0;
+  vector<EdgeId> edge_ids;
+};
+
+std::ostream& operator<<(std::ostream& out, const Vertex& vertex) {
+  out << "{" << endl;
+  out << "      \"id\": " << vertex.id << "," << endl;
+  out << "      \"edge_ids\": [" << vertex.edge_ids << "]" << endl;
+  out << "    }";
```

Comment:
nit: Скобочки ведь закрываются на том же уровне, что и открываются. Зачем эти пробелы? Эта функция не знает о том, сколько отступов уже в файле, и предполагать она этого не должна. Если хочешь форматировать финальный результат, это должно быть сделано, по хорошему, каким-нибудь другим способом.
```suggestion
  out << "}";
```